src/components/TabsMainPage.tsx

```diff
@@ -0,0 +1,222 @@
+        title: localization.main.tabsText,
+        query: '',
+        response: '',
+        variables: '',
+        active: false,
+      };
+      setTabs((prevState) => {
+        return prevState.concat(tab);
+      });
+      setTabs((prevState) => {
+        return prevState.map((t) =>
+          t.id === tab.id ? { ...t, active: true } : { ...t, active: false }
+        );
+      });
+      setQuery('');
+      setResponse('');
+      setVariables('');
+    }
+  };
+
+  useEffect(() => {
+    setTabs((prevState) => {
+      return prevState.map((t) => (t.active ? { ...t, query, response, variables } : { ...t }));
+    });
+  }, [query, response, variables]);
+
+  const setActiveButton = (e: React.MouseEvent<HTMLDivElement, MouseEvent>, tab: ITab) => {
+    e.stopPropagation();
+    setQuery(tab.query);
+    setResponse(tab.response);
+    setVariables(tab.variables);
+
+    setTabs((prevState) => {
+      return prevState.map((t) =>
+        t.id === tab.id ? { ...t, active: true } : { ...t, active: false }
+      );
+    });
+  };
+
+  const deleteTab = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>, tab: ITab) => {
+    e.stopPropagation();
+
+    if (tabs.length < 2) {
+      showMsg({ type: 'error', content: localization.main.tabsMinLimits });
+      return;
+    }
+
+    const index = tabs.findIndex((t) => t.id === tab.id);
+
+    let newTab: ITab;
+
+    if (tab.active) {
+      if (index === 0 && tabs.length > 1) {
+        setTabs((prevState) => {
+          return prevState.map((t, i) =>
+            i === index + 1 ? { ...t, active: true } : { ...t, active: false }
+          );
+        });
+        newTab = tabs[index + 1];
+      } else {
+        setTabs((prevState) => {
+          return prevState.map((t, i) =>
+            i === index - 1 ? { ...t, active: true } : { ...t, active: false }
+          );
+        });
+
+        newTab = tabs[index - 1];
+      }
+
+      setQuery(newTab.query);
+      setResponse(newTab.response);
+      setVariables(newTab.variables);
+    }
+
+    setTabs((prevState) => {
+      return prevState.filter((el) => el.id !== tab.id);
+    });
+  };
+
+  return (
+    <div>
+      <Stack
+        direction="row"
+        justifyContent="space-between"
+        width="100%"
+        position="relative"
+        style={{ marginBottom: '10px' }}
+      >
+        <Stack
+          direction="row"
+          justifyContent="start"
+          width="80%"
+          gap={1}
+          position="relative"
+          style={{ overflowX: 'auto', scrollbarWidth: 'none' }}
+        >
+          {tabs.map((t) => (
+            <TabButton
+              key={t.id}
+              onClick={(e) => setActiveButton(e, t)}
+              style={{
+                color: t.active ? indigo[700] : blueGrey[200],
+                borderBottom: `2px solid ${t.active ? indigo[700] : ''}`,
+              }}
+            >
+              <EditableText initText={t.title} active={t.active} setTabs={setTabs} />
```

Comment:
![image](https://github.com/shish-ko/graphiql-app/assets/55231237/a4acbc91-378a-4b97-9d31-578e66418f8f)
Don't allow to set empty tab title.